spec/discovery.md

```diff
@@ -59,6 +59,18 @@ curl $(echo "$urltmpl" | sed -e "s/{name}/$name/" -e "s/{version}/$version/" -e
 
 where _name_, _version_, _os_, and _arch_ are set to their respective values for the image, and _ext_ is either `aci` or `aci.asc` for retrieving an App Container Image or signature respectively.
 
+The client MUST accept only best matching templates. Best matching templates are the one where all the templates labels can be substituted and with the highest number of substituted labels.
```

Comment:
> If all client-supplied labels must be substituted

That was the original idea in #580, in this patch there isn't the requirement to substitute all the client supplied labels but to return only the templates that match the highest number of client-supplied labels.

Perhaps, If you agree on this new idea, the spec description needs your help for better wording :smile: